ui/src/helpers/get-defaults.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2024 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+// Search the secrets hierarchically for default values.
+
+// TypeScript gets confused when accessing nested properties, so we ensure type safety
+// of the input of updateBaseDefaults() while letting 'any' types for internal processing.
+
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+
+import { PagedResponse_Secret_ } from '@/api/requests';
+import { CreateRunFormValues } from '@/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/create-run';
+
+/**
+ * Get the default values from the secrets.
+ *
+ * @param orgSecrets The organization secrets.
+ * @param prodSecrets The product secrets.
+ * @param repoSecrets The repository secrets.
+ * @returns The default values as a map.
+ */
+function getDefaultValues(
+  orgSecrets: PagedResponse_Secret_,
+  prodSecrets: PagedResponse_Secret_,
+  repoSecrets: PagedResponse_Secret_
+): { [key: string]: string } {
+  const defaultValuesMap: { [key: string]: string } = {};
+
+  // Helper function to add secrets to the map
+  const addSecrets = (secrets: PagedResponse_Secret_) => {
+    secrets.data.forEach((secret) => {
+      if (secret.name.endsWith('DEFAULT') && secret.description) {
```

Comment:
To recap what @haikoschol proposed during our internal discussion:
- any secret names that have special semantic should start with a "_" as it's a syntactically valid character that hardly any real secret name would start with (in our discussion we were talking about a dash instead, but Scaleway does not seem to allow that as secret names need to match the regex `^[_a-zA-Z0-9]([-_.a-zA-Z0-9]*[_a-zA-Z0-9])?$`)
- instead of using "DEFAULT" as a suffix, we should use it as a prefix following the leading "_"
- maybe we should also introduce an infix that further describes *what for* the default is, like "PARAMETER" or "PARAM"

So as an example, the name of a secret whose value contains the default value for the parameter identified by the `jobConfig.evaluator.ruleSet` JSON path would be named `_DEFAULT_PARAM-jobConfig.evaluator.ruleSet`.

Reply:
Back-end only allows letters, numbers, hyphens and spaces in secret names, no underscores.

Reply:
> no underscores.

Let's double-check on that as I'm not aware of any generic check for underscores in the code. IIRC any restrictions should be secret-provider-specific, and at least Scaleway explicitly allowed underscores when I implemented support for it.

Reply:
Ah, it's this validation:

https://github.com/eclipse-apoapsis/ort-server/blob/a211137352b46dc3658fa954f18483233db95694/api/v1/model/src/commonMain/kotlin/Secret.kt#L55-L57